source/Irrlicht/COpenGLDriver.cpp

```diff
@@ -2402,19 +2453,27 @@ void COpenGLDriver::setRenderStates3DMode()
 		ResetRenderStates = true;
 	}
 
+    //actually i'm commenting-out ALMOST all in this if-statemenet
 	if (ResetRenderStates || LastMaterial != Material)
 	{
 		// unset old material
 
-		if (LastMaterial.MaterialType != Material.MaterialType &&
+		/* TODO what OnUnsetMaterial() is responsible for? Can i remove this?
+        if (LastMaterial.Pipeline != Material.Pipeline &&
 				static_cast<uint32_t>(LastMaterial.MaterialType) < MaterialRenderers.size())
-			MaterialRenderers[LastMaterial.MaterialType].Renderer->OnUnsetMaterial();
```

Comment:
I think it just called the appropriate call-back... see the definitions of `OnUnsetMaterial`